tests/Integration/ProductHub.IntegrationTests/ProductControllerIntegrationTests.cs

```diff
@@ -1,10 +1,123 @@
+using AutoFixture;
+using FluentAssertions;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Newtonsoft.Json;
+using ProductHub.Domain.Product;
+using System.Net;
+using System.Net.Http.Json;
+using System.Text;
+
 namespace ProductHub.IntegrationTests;
 
 public class ProductControllerIntegrationTests
+    : IClassFixture<WebApplicationFactory<Program>>
 {
+    private readonly Fixture _fixture;
+    private readonly HttpClient _httpClient;
+    private readonly WebApplicationFactory<Program> _appFactory;
+
+    public ProductControllerIntegrationTests()
+    {
+        _appFactory = new WebApplicationFactory<Program>();
+        _httpClient = _appFactory.CreateClient();
+        _fixture = new Fixture();
+    }
+
+    [Fact]
+    public async Task GetProductsAsync_Success_EndpointReturnsOkResult()
+    {
+        // Arrange
+        var expectedProducts = _fixture.CreateMany<Product>();
+
+        // Act
```

Comment:
Why are you specifically putting port and local host. Use test host as base url.

Reply:
Okay will add a utility file to fix this.